### ISSUE TITLE
Using a chosen-element always fires a horizontal page scroll to the left

### ISSUE THREAD

**Issue by claudio10** [harvesthq/chosen#1410]
Didn't find this issue along the others already reported. It's not really a bug, but an unexpected behaviour. 

Summary of the issue:
When having a viewport that needs horizontal scroll bars, and a chosen-element is set rightoutside that viewport, the user will have to scroll the page to the right, until he can see the chosen-element he wants to use/select. Selecting it or selecting an li-item will result in a total scroll back left of the page, putting the just selected chosen-element outside of that viewport again.

Expected behaviour: The viewport should just keep on the selected element, without moving/scrolling.

Tracked OS: 
WinVista: FireFox 22.0, Chrome Version 28.0, Safari 5.1.7, IE 9.0
Android 4.1.2: Default Android Browser & Chrome (can't see where to get BrowserVers)

Steps required to reproduce the bug:

1. Go to http://harvesthq.github.io/chosen/

2. Minimize the browsers viewport until you see the page with bottom scroll bars like this:
![image](https://f.cloud.github.com/assets/5096188/862117/e6148146-f5e5-11e2-84de-1eece0569d04.png)

3. Scroll to the right until you can see the right chosen-element like this:
![image](https://f.cloud.github.com/assets/5096188/862122/2edaf2ac-f5e6-11e2-8424-a2f58c09d4e6.png)

4. Select a option of that second chosen-element and you will see the page being scrolled totally back to the left, putting the just selected element outside the viewport again.

The behaviour will occure whenever the page has to be scrolled right for a chosen-selection, e.g. wide-screen pages with multiple side by side chosen-elements, mobile devices/tablets with reduced viewports. 

Best wishes and thx for your great solution & contribution!!!

Claudio

**Comment by Mike0609** [comment]
Fix by css
34 line - .chosen-container .chosen-drop
change 

left: -9999px;
to
left: 0;

and add display: none;

46 line - .chosen-container.chosen-with-drop .chosen-drop
replace: 

left: 0;
to
display: block;

and everything works fine

**Comment by qwelyt** [comment]
@Mike0609 That solution does not seem to work. Any select that does not have a searchbar produces the left-bug.

**Comment by Mike0609** [comment]
https://github.com/qwelyt I'm sorry, I checked and found inaccuracies, here is a link to a working example: http://chirskiy.ru/chosen_fix/

**Comment by qwelyt** [comment]
@Mike0609 As far as our product goes, that works wonders! Thanks!
This should be merged upstream.
It seems to work on both desktop and mobile devices. So big thanks!

**Comment by qwelyt** [comment]
@Mike0609 At further inspection, adding display:none; to ``.chosen-container .chosen-drop`` makes tabbing to an element impossible. Meaning you can no longer work from the keyboard alone, which is handy with big forms.

**Comment by rashmirao05** [comment]
Hello,

I don't see the fix for this in the latest release . Has this been not incorporated?
I faced this issue (Horizontal shift)  for a long time in my project.

Thanks,
Rashmi.

**Comment by rashmirao05** [comment]
@claudio10 @gorner Hi guys I faced this issue for a long time in the project im currently working in.
So whats the ultimate solution.?

I do not see the resolution incorporated in the latest version of chosen plugin.
I set the method this.search_field.focus(); to comments and it solves the issue for me.
So is this the final resolution?

Please suggest.
Appreciate your help.
Rashmi.

**Comment by tjschuck** [comment]
No solution, but just confirming that this still seems to be a valid issue in Chosen v1.5.1 (the current release version).

**Comment by rashmirao05** [comment]
@tjschuck thank u :) true that this issue still exists in the current release.

**Comment by billrob** [comment]
Hey guys, great tool here.  I confirmed the behavior still exists under IE11 on the demo site.  Do you think we could add an "In ViewPort" check to call focus?

**Comment by FleKr** [comment]
We got rid of the problem with the following css:
```css
.chosen-container .chosen-drop {
	left:0;
	opacity:0;
	height:0;
	overflow: hidden;
}
.chosen-container.chosen-with-drop .chosen-drop {
	opacity:1;
	height:auto;
}
```
Because opacity is not supported by all browsers, there will by a 1px shadow-line beneath the chosen-select. We tried it with Firefox 47 and IE11, both work fine.

Short explanation: 
We disable the left:-9999 so the focus() event is to going to scroll anywhere. `display: none;` would disable tabbing-support, so we do not hide the element, but make it invisible by height:0 and so that the 1px shadow disappears we use opacity: 0 but this is not supported by all browsers so they have to deal with 1px shadow… 

For me, this solved the issue, but i did not test it with mobile devices.
